pkg/reconciler/route/traffic/rollout.go

```diff
@@ -70,3 +73,100 @@ type RevisionRollout struct {
 	// target percentage.
 	Percent int `json:"percent"`
 }
+
+// Step processes previous Rollout object and compares to the current
+// rollout state. If there is different, Step will start or stop the rollout
+// and updates `RevisionRollout` objects of `cur` accoringly.
+// At the end of the call `cur` contains the desired traffic shape.
+// Step returns `true` if any changes have been made.
+func (cur *Rollout) Step(prev *Rollout) bool {
```

Comment:
My understanding is that "current" is the goal state of the rollout.  In fact I'd probably characterize "previous" as the "current rollout state", which is confusing 😬